components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/MessageProcessorResource.java

```diff
@@ -0,0 +1,252 @@
+        Map<String, MessageProcessor> processorMap = synapseConfiguration.getMessageProcessors();
+        JSONObject jsonBody = Utils.createJSONList(processorMap.size());
+        processorMap.forEach((key, value) -> addToJSONList(value, jsonBody.getJSONArray(Constants.LIST)));
+        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+    }
+
+    /**
+     * Create JSON response with information related a defined message processor
+     * @param messageContext synapse message context
+     * @param name name of the message processor
+     * */
+    private void populateMessageProcessorData(MessageContext messageContext, String name) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        MessageProcessor messageProcessor = synapseConfiguration.getMessageProcessors().get(name);
+        if (Objects.nonNull(messageProcessor)) {
+            Utils.setJsonPayLoad(axis2MessageContext, getMessageProcessorAsJson(messageProcessor));
+        } else {
+            Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Message processor does not exist"));
+        }
+    }
+
+    /**
+     * Adds the specified message processor to a JSON list
+     * @param messageProcessor
+     * @param processorList JSON list of processors
+     * */
+    private void addToJSONList(MessageProcessor messageProcessor, JSONArray processorList) {
+        JSONObject messageProcessorObject = new JSONObject();
+        messageProcessorObject.put(Constants.NAME, messageProcessor.getName());
+        messageProcessorObject.put(TYPE_PROPERTY, getMessageProcessorType(messageProcessor));
+        messageProcessorObject.put(STATUS_PROPERTY, getProcessorState(messageProcessor.isDeactivated()));
+        processorList.put(messageProcessorObject);
+    }
+
+    /**
+     * Returns the JSON representation of a message processor
+     *
+     * @param messageProcessor
+     * @return JSONObject MessageProcessor
+     **/
+    private JSONObject getMessageProcessorAsJson(MessageProcessor messageProcessor) {
+        JSONObject jsonBody = new JSONObject();
+        jsonBody.put(Constants.NAME, messageProcessor.getName());
+        jsonBody.put(TYPE_PROPERTY, getMessageProcessorType(messageProcessor));
+        jsonBody.put(CONTAINER_PROPERTY, messageProcessor.getArtifactContainerName());
+        jsonBody.put(FILE_NAME_PROPERTY, messageProcessor.getFileName());
+        jsonBody.put(PARAMETER_PROPERTY, messageProcessor.getParameters());
+        jsonBody.put(MESSAGE_STORE_PROPERTY, messageProcessor.getMessageStoreName());
+        jsonBody.put(STATUS_PROPERTY, getProcessorState(messageProcessor.isDeactivated()));
+
+        return jsonBody;
+    }
+
+    /**
+     * Returns the type of the message processor
+     * @param messageProcessor
```

Comment:
Shall we add a description? This is shown as a warning in the IDE.

Reply:
done